tools/pythonpkg/tests/fast/spark/test_spark_group_by.py

```diff
@@ -116,6 +116,22 @@ def test_group_by(self, spark):
             == "[Row(department='Finance', sum_salary=351000, avg_salary=87750.0, sum_bonus=81000, max_bonus=24000), Row(department='Sales', sum_salary=257000, avg_salary=85666.66666666667, sum_bonus=53000, max_bonus=23000)]"
         )
 
+        df = spark.createDataFrame(
```

Comment:
If possible, try to create separate methods for tests, if this were to fail in the future it would be difficult to debug as that work to separate out the relevant parts would need to be done afterwards, but for now let's leave this as is :+1:

Reply:
I got the impression from some of the existing tests that the preference is towards not separating it too much but totally agree, I also prefer smaller self-contained tests. Thanks for the feedback, will do!